### PR TITLE
Upgrade RN to 0.62.2

### PR DESCRIPTION
**Pull request by marecar3**
Update React Native to 0.62.0

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/19950
WordPressAndroid: https://github.com/wordpress-mobile/WordPress-Android/pull/11217
WordPressiOS: https://github.com/wordpress-mobile/WordPress-iOS/pull/13752
react-native-video PR: https://github.com/wordpress-mobile/react-native-video/pull/5/
react-native-svg PR: https://github.com/wordpress-mobile/react-native-svg/pull/10

To test: Gutenberg Editor should work as expected. 
Try to run manual tests from the list.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -43,11 +63,21 @@ target 'gutenberg' do
...
-    inherit! :search_paths
+    inherit! :complete
     # Pods for testing
   end
 
   use_native_modules!
+
+  # Enables Flipper.
+  #
+  # Note that if you have use_frameworks! enabled, Flipper will not work and
```
Comment (marecar3) [inline ios/Podfile:74]: Hey @SergioEstevao 👋 
please, can you take a look at this and see if we can enable Flipper? 
(as it seems that we have use_frameworks! enabled?) Thanks.

File: ios/Podfile [hunk 2]
```diff
@@ -43,11 +63,21 @@ target 'gutenberg' do
   pod 'Gutenberg', :path => '../Gutenberg.podspec'
 
   target 'gutenbergTests' do
-    inherit! :search_paths
+    inherit! :complete
```
Comment (marecar3) [inline ios/Podfile:66]: @SergioEstevao I have saw this change when I used this tool: https://react-native-community.github.io/upgrade-helper/?from=0.61.5&to=0.62.0
not sure if we need it or not, please re-check if you can, thanks!

**Comment by marecar3** [comment]
This is the list of Gutenberg manual tests that we should follow during testing. 

# Gutenberg tests

## Writing flow:

- [ ] Block insertion - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/block-insertion.md)
- [ ] Copy paste - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/copy-paste.md)
- [ ] Multilane components - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/multiline-components.md)
- [ ] Rich text formatting - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/rich-text-formatting.md)
- [ ] Splitting and merging - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/splitting-merging.md)
- [ ] Undo redo - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/undo-redo.md)

## Other:

### Button
- [ ] Button block - Render custom text color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc011)
- [ ] Button block - Render custom background color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc010)
- [ ] Button block - Render gradient background color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc012)
- [ ] Button block - Check if selection / caret color matches font color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc015)
- [ ] Button block - Edit text styles - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc003)
- [ ] Button block - Fallback to default colors in case theme colors are used - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc013)
- [ ] Button block - New Button is created with the gray color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc014)
- [ ] Button block - Settings: Change Button border radius - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc009)
- [ ] Button block - The newly created button has set background color to the theme-default color in preview - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc016)
- [ ] Button block - Tapping `delete` key removes block when Button is empty - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc017)
- [ ] Button block - Settings: Open in new tab - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc005)
- [ ] Button block - Settings: Link rel - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc006)
- [ ] Button block - Settings: Link URL - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc004)
- [ ] Button block - Settings: Remove link - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc008)
- [ ] Button block - Settings: Synchronize with button options - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc007)
- [ ] Button block - Link from the clipboard is automatically added into the empty URL field in link settings and button options - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc002)
- [ ] Button block - Toolbar link button is active when Button has link - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc018)
- [ ] Button block - Button max width is calculated OK inside inner blocks(iOS only) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc001)

### Cover
- [ ] Cover - Add Video Background from WordPress Media library - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/cover.md#tc005)
- [ ] Cover - Replace media - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/cover.md#tc006)
- [ ] Cover - Gradient background is rendered properly - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/cover.md#tc007)
- [ ] Cover - Gradient overlay is rendered properly - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/cover.md#tc008)

### DarkMode
- [ ] Gallery block - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc001)
- [ ] Shortcode block - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc002)
- [ ] Media Text block - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc003)
- [ ] Spacer block - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc004)
- [ ] Button block - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc005)
- [ ] Group - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc006)

### Gallery
- [ ] Gallery block - Close/Re-open post with an ongoing image upload - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc001) (Note: @hypest reported success on this test. [See here](https://github.com/wordpress-mobile/WordPress-Android/pull/11569#issuecomment-607739703).)
- [ ] Gallery block - Close post with an ongoing image upload - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc002)
- [ ] Gallery block - Insert image from device (failing) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc006)
- [ ] Gallery block - Insert image from device (cancel) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc009)
- [ ] Gallery block - Add Gallery Caption - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc003)
- [ ] Gallery block - Add Gallery Image Caption - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc004)
- [ ] Gallery block - Try adding same images from WP Media library and moving the images around - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc010)
- [ ] Gallery block - Settings: Link to - steps
- [ ] Gallery block - Settings: Column number - steps
- [ ] Gallery block - Settings: Crop images - steps

### Group
- [ ] Group - AppenderButton is rendered - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc001)
- [ ] Group - Deep nesting is possible (iOS only) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc002)
- [ ] Group - Check if Group placeholder is visible for the unselected state - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc003)
- [ ] Group - Check if Group placeholder is render in nested structure - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc004)
- [ ] Group - Nested block have proper border styling - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc005)
- [ ] Group - Nested block have proper margins values - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc006)
- [ ] Group - Nested selection cause applying dimmed style on the rest of blocks - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc007) - This feature is removed
- [ ] Group - Breadcrumbs on FloatingToolbar is properly displayed - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc008) - With one exception - we turned off the floating toolbar for root level blocks that are not inner blocks.
- [ ] Group - Navigation up button works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc009)
- [ ] Group - Navigation down works according to parent-first approach - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc010)
- [ ] Group - Cross navigation between blocks works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc011) (Note: @hypest had success in this test, [see here](https://github.com/wordpress-mobile/WordPress-Android/pull/11569#issuecomment-607752743))
- [ ] Group - Ungroup button works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc012)
- [ ] Group - Check if in DarkMode all components gets proper colors (iOS) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc013)
- [ ] Group - Check if nested Placeholder block can be replaced - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc014)

### Image
- [ ] Image [TC001](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/image.md#tc001)
- [ ] Image [TC002](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/image.md#tc002)
- [ ] Image [TC003](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/image.md#tc003)
- [ ] Image [TC004](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/image.md#tc004) (unable to return fast enough to see loading bar in the post, but otherwise this works fine)
- [ ] Image [TC005](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/image.md#tc005)

### Media interaction

- [ ] Media interaction - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-interaction.md)

### Media text

- [ ] Media interaction - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-text.md)

### Spacer

- [ ] Spacer block - Spacer is rendered without crash - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc001)
- [ ] Spacer block - Settings: Control spacer height using the slider - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc002)
- [ ] Spacer block - Settings: Control spacer height using the text input - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc003)
- [ ] Spacer block - Settings: Available height range is correct - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc004)
- [ ] Spacer block - Settings: Height range extends if Spacer comes from the web is higher than 500px - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc005)
- [ ] Spacer block - Spacer in horizontal layout works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc006)

### Video

- [ ] Video [TC001](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/video.md#tc001)
- [ ] Video [TC002](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/video.md#tc002)
- [ ] Video [TC003](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/video.md#tc003) (Caption alignment seems a little funky in the published post, seems the same as from Web gutenberg though)
- [ ] Video [TC004](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/video.md#tc004)
- [ ] Video [TC005](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/video.md#tc005)
- [ ] Video [TC006](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/video.md#tc006)

File: ios/Podfile (hunk 1 above)
Comment (SergioEstevao) [reply ios/Podfile:74]: Last time I tried on the beta versiosn we couldn't, I can give it a go again if you want?

Comment (marecar3) [reply ios/Podfile:74]: Well if you think that it doesn't make a sense then we can leave it as it is and try in the future or if there is some other way to enable it?

**Comment by pinarol** [comment]
I am changing the milestone of this PR as 1.27 because the 1.26 release branch will be cut quite soon. Let me know if you instead wanted to include this to 1.26.

**Comment by guarani** [comment]
Tested on Android Galaxy S10

### Video
- [ ] TC001 to TC006

**Comment by marecar3** [comment]
> Tested on Android Galaxy S10
> 
> ### Image
> * ✅ TC001 to TC005
> * ⚠️ [TC006](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/image.md#tc006) I noticed an _existing_ bug here: Add an image to an image block | switch to the classic editor | tap the image to view image settings | switch back to block editor | observe "Problem displaying block message" (or red screen in debug builds). **This is existing bug and is an edge case, so does not block this PR at all.** I'll file a separate bug report for this.
> 
> ### Media & Text
> * ✅ TC001 to TC007
> 
> ### Spacer
> * ✅ TC001 to TC006
> 
> Note: I don't see the option on the web to set the spacer to height > 500px, so not sure how to interpret [TC005](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc005)
> 
> ### Video
> * ✅ TC001 to TC006

Thanks for the great effort @guarani!

**Comment by SergioEstevao** [comment]
I'm on this for iOS

## Writing flow:

- [ ] Block insertion - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/block-insertion.md)
- [ ] Copy paste - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/copy-paste.md)
- [ ] Multilane components - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/multiline-components.md)
- [ ] Rich text formatting - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/rich-text-formatting.md)
- [ ] Splitting and merging - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/splitting-merging.md)
- [ ] Undo redo - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/writing-flow/undo-redo.md)

**Comment by mchowning** [comment]
👋 @guarani ! Thanks so much for testing this!

> Note: I don't see the option on the web to set the spacer to height > 500px, so not sure how to interpret TC005

You can probably achieve that by switching to the html, manually editing the spacer height in the html for the spacer block, and then switching back to the visual editor.

**Comment by mkevins** [comment]
Hi @marecar3 :wave: :smile: I see this PR has been approved last week, but I wasn't sure if there are other any other blockers. I'll bump the milestone to `1.28`, but please let me know if you want this one in for the `1.27` release.

**Comment by lukewalczak** [comment]
Hello @marecar3, we cannot remove the whole react-native patch (_part related to shadow view can be deleted since these changes are already merged into react-native_), along with `patch-package` lib and script, because recently I had to add [new patch](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2220/files#diff-6c5ad6f771a45ea3b93efab244996f4eR1) for disabling buttons in action sheet. The same [patch](https://github.com/wordpress-mobile/WordPress-iOS/pull/14062) is applied in iOS app. 

I can create a commit with patching to 0.62 if you would like to :v:

**Comment by SergioEstevao** [comment]
This new patch comes from what version of React Native?

**Comment by lukewalczak** [comment]
> This new patch comes from what version of React Native?

0.61.5 - the current version on gb-mobile

**Comment by marecar3** [comment]
> Hello @marecar3, we cannot remove the whole react-native patch (_part related to shadow view can be deleted since these changes are already merged into react-native_), along with `patch-package` lib and script, because recently I had to add [new patch](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2220/files#diff-6c5ad6f771a45ea3b93efab244996f4eR1) for disabling buttons in action sheet. The same [patch](https://github.com/wordpress-mobile/WordPress-iOS/pull/14062) is applied in iOS app.
> 
> I can create a commit with patching to 0.62 if you would like to ✌️

Hey @lukewalczak, thanks for the ping.
Yes, please if you can, revert back patch with changes that we need for RN `0.62.2`

**Comment by lukewalczak** [comment]
Between the versions 0.61.5 and 0.62.2 they've changed the structure a bit, so I have to apply the new patch. Will handle it!

**Comment by marecar3** [comment]
> Between the versions 0.61.5 and 0.62.2 they've changed the structure a bit, so I have to apply the new patch. Will handle it!

Thanks a lot for helping with that one!

**Comment by mchowning** [comment]
We're preparing to cut the 1.30 release branches, so I'm going to bump this to 1.31. If this needs to make it in the 1.30 release though, just let me know. 🙂

**Comment by SergioEstevao** [comment]
Closing this PR in favor of: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2439 that is based on the monorepo work.

**Pull request closed without merging**